pkg/blockdevice/v1alpha1/blockdevice.go

```diff
@@ -356,3 +356,29 @@ func (bdl *BlockDeviceList) GetBlockDevice(bdcName string) *ndm.BlockDevice {
 	}
 	return nil
 }
+
+// CheckBlockDeviceDeviceID compare the device Ids present in block device
+// with provided list
+func (bdl *BlockDeviceList) CheckBlockDeviceDeviceID(bdDevIDList []string) error {
+	if len(bdl.Items) == 0 || len(bdDevIDList) == 0 {
+		return errors.Errorf("csp deviceIDList or csp blockdevice list is empty")
+	}
+	if len(bdl.Items) != len(bdDevIDList) {
+		return errors.Errorf("csp deviceIDList length is not matched with csp blockdevice list length")
+	}
+	for i, bdcObj := range bdl.Items {
+		bdcObj := bdcObj
+		customBD := &BlockDevice{
+			BlockDevice: &bdcObj,
+		}
+		bdID := customBD.GetDeviceID()
+		if bdID != bdDevIDList[i] {
```

Comment:
this seems to be comparing the order as well.. do we need to match the order also?

Reply:
Yes, Since we are reading both of BDDevID and BDName in same order we can do it.